### PR TITLE
New version: LuculentSystems.AltA2DP version 1.2.0

### PR DESCRIPTION
**Pull request by LuculentSystems**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-pkgs/pull/109464&drop=dogfoodAlpha

### CODE REVIEW

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Created with YamlCreate.ps1 v2.2.4 $debug=NVS0.CRLF.7-2-10.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: LuculentSystems.AltA2DP
+PackageVersion: 1.2.0
```
Comment (ItzLevvie) [inline manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml:5]:
```suggestion
PackageVersion: 1.2.0.1
```
This must match Control Panel.

You also need to change `manifests/l/LuculentSystems/AltA2DP/1.2.0` to `manifests/l/LuculentSystems/AltA2DP/1.2.0.1`

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Created with YamlCreate.ps1 v2.2.4 $debug=NVS0.CRLF.7-2-10.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultLocale.1.4.0.schema.json
+
+PackageIdentifier: LuculentSystems.AltA2DP
+PackageVersion: 1.2.0
```
Comment (ItzLevvie) [inline manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.locale.en-US.yaml:5]:
```suggestion
PackageVersion: 1.2.0.1
```
This must match Control Panel.

You also need to change `manifests/l/LuculentSystems/AltA2DP/1.2.0` to `manifests/l/LuculentSystems/AltA2DP/1.2.0.1`

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Created with YamlCreate.ps1 v2.2.4 $debug=NVS0.CRLF.7-2-10.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.4.0.schema.json
+
+PackageIdentifier: LuculentSystems.AltA2DP
+PackageVersion: 1.2.0
```
Comment (ItzLevvie) [inline manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.yaml:5]:
```suggestion
PackageVersion: 1.2.0.1
```
This must match Control Panel.

You also need to change `manifests/l/LuculentSystems/AltA2DP/1.2.0` to `manifests/l/LuculentSystems/AltA2DP/1.2.0.1`

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml (hunk 1 above)
Comment (LuculentSystems) [reply manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml:5]: Thank you for your suggestion, but we've been using 3-segment version number for all public releases.
And, correct me if I'm wrong, but there's no WinGet requirement about how to use the PackageVersion.

Comment (ItzLevvie) [reply manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml:5]: > And, correct me if I'm wrong, but there's no WinGet requirement about how to use the PackageVersion.

In https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.4.0/installer.md#fields for `PackageVersion` it says
- The current best practice is to use the value reported in Add / Remove Programs when this version of the package is installed.
- This practice may seem contrary to using semantic versioning, but it provides the best end to end experience for customers.

^ Add & Remove Programs (ARP) is Control Panel (Win32) / Installed apps (UWP).

It is possible to use `PackageVersion: 1.2.0` and use `DisplayVersion: 1.2.0.1` as part of AppsAndFeaturesEntries but I believe you will have to do that for all previous versions submitted to the WinGet package repository.

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
See https://github.com/microsoft/winget-pkgs/pull/109464#discussion_r1225963927

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.yaml (hunk 3 above)
Comment (LuculentSystems) [reply manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.yaml:5]: Could you clarify where that requirement comes from?

**Comment by SpecterShell** [comment]
The `PackageVersion` field must match the version that the installer writes to the registry during installation, with which WinGet  identifies which version has been installed in the system for upgrading purpose.

image

**Comment by LuculentSystems** [comment]
Hello @SpecterShell 
Thank you for your reply, but again, where does that "requirement" comes from?
As far as I know, the schema doesn't define anything about the usage of the package version.  If there's any documentation (not your word), it would be appreciated.

**Comment by SpecterShell** [comment]
https://github.com/microsoft/winget-pkgs/pull/109464#discussion_r1225963927 🤔

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.locale.en-US.yaml (hunk 2 above)
Comment (stephengillie) [reply manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.locale.en-US.yaml:5]: Please update file name and path to match this change.

File: manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml (hunk 1 above)
Comment (stephengillie) [reply manifests/l/LuculentSystems/AltA2DP/1.2.0/LuculentSystems.AltA2DP.installer.yaml:5]: Please update file path to match this change.

**Comment by stephengillie** [comment]
To prevent a manifest validation error, when updating the PackageVersion, please also update the manifest paths in the file system. This is because the PackageVersion is checked against the version section of the path, and validation will fail if they don't match.

**Pull request closed without merging**